### DISCUSSION TITLE
Is there a way to restore the object when the hardware fails?

### DISCUSSION THREAD

**Discussion by dunghn87** [minio/minio#15009] [Q&A] [answered]
Is there a way to restore the object when the hardware fails?
Here my cluster http://minio{1...4}/data/disk{1...4} http://minio{5...8}/data/disk{1...4}

minio1: not found
minio2: /data/disk{1..4}/butket/objectname/xl.meta
minio3: /data/disk{1..4}/butket/objectname/xl.meta
minio4: not found

minio5: not found
minio6: not found
minio7: not found
minio8: not found

I Lost Quorum (server minio1, 4) then add new server 1,4 (with disk{1...4})
MinIO VERSION 2022-02-05T04:40:59Z

_Originally posted by @dunghn87 in https://github.com/minio/minio/discussions/14717#discussioncomment-2853244_

**Comment by klauspost** [comment, marked as answer]
Upgrade.

The object should be readable with normal operations. The objects will eventually be healed if recoverable or deleted if unrecoverable.

**Reply by dunghn87** [reply]
I run
./mc admin heal minio/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg
ba-cam
    0/0 objects; 0 B in 1s
    ┌────────┬───┬─────────────────────┐
    │ Green  │ 3 │ 100.0% ████████████ │
    │ Yellow │ 0 │   0.0%              │
    │ Red    │ 0 │   0.0%              │
    │ Grey   │ 0 │   0.0%              │
    └────────┴───┴─────────────────────┘
is object recoverable? how to check if object normal

**Reply by klauspost** [reply]
You can try to download it I presume. Looks good.

**Reply by dunghn87** [reply]
cannot download object.
here link my object https://minio.bagroup.vn/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg
MinIO VERSION 
2022-02-05T04:40:59Z

**Reply by klauspost** [reply]
Your object probably lost quorum.

Can you post output of `mc support inspect ALIAS/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/**` (replace ALIAS) I can take a look.

**Reply by dunghn87** [reply]
./mc support inspect minio/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/**
mc: File data successfully downloaded as inspect.0886ba56.zip

Here file zip
[inspect.0886ba56.zip](https://github.com/minio/minio/files/8825897/inspect.0886ba56.zip)

**Reply by klauspost** [reply]
@dunghn87 You have lost data on 7 of 16 disks for this object. There are 8 shards left, but 12 is needed.

Your object is lost.

**Reply by mshanmu** [reply]
@klauspost  , I assume you intended to say "You have lost data on 8 of 16 disks for this object"?

**Reply by klauspost** [reply]
No. There is 9 shards left.

**Reply by klauspost** [reply]
The object is written with ec:4

**Reply by mshanmu** [reply]
`$ ./xl-meta ~/inspect.0886ba56.zip 
{
	"minio2:9000/data/disk1/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/xl.meta": {"Versions":[{"Idx":0,"Header":{"VersionID":"00000000000000000000000000000000","ModTime":"2022-05-26T02:48:44.625903892-04:00","Signature":"b63cd20e","Type":1,"Flags":6},"Metadata":{"Type":1,"V2Obj":{"ID":"AAAAAAAAAAAAAAAAAAAAAA==","DDir":"D9JTg2uzSz+sKxN16Z7e1Q==","EcAlgo":1,"EcM":12,"EcN":4,"EcBSize":1048576,"EcIndex":7,"EcDist":[6,7,8,9,10,11,12,13,14,15,16,1,2,3,4,5],"CSumAlgo":1,"PartNums":[1],"PartETags":null,"PartSizes":[103794],"PartASizes":[103794],"Size":103794,"MTime":1653547724625903892,"MetaSys":{"x-minio-internal-inline-data":"dHJ1ZQ=="},"MetaUsr":{"etag":"71f3fcf7d01af8a9f1483ad512069e03","content-type":"image/jpg"}}}}]}
,
	"minio3:9000/data/disk1/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/xl.meta": {"Versions":[{"Idx":0,"Header":{"VersionID":"00000000000000000000000000000000","ModTime":"2022-05-26T02:48:44.625903892-04:00","Signature":"b63cd20e","Type":1,"Flags":6},"Metadata":{"Type":1,"V2Obj":{"ID":"AAAAAAAAAAAAAAAAAAAAAA==","DDir":"D9JTg2uzSz+sKxN16Z7e1Q==","EcAlgo":1,"EcM":12,"EcN":4,"EcBSize":1048576,"EcIndex":8,"EcDist":[6,7,8,9,10,11,12,13,14,15,16,1,2,3,4,5],"CSumAlgo":1,"PartNums":[1],"PartETags":null,"PartSizes":[103794],"PartASizes":[103794],"Size":103794,"MTime":1653547724625903892,"MetaSys":{"x-minio-internal-inline-data":"dHJ1ZQ=="},"MetaUsr":{"etag":"71f3fcf7d01af8a9f1483ad512069e03","content-type":"image/jpg"}}}}]}
,
	"minio2:9000/data/disk2/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/xl.meta": {"Versions":[{"Idx":0,"Header":{"VersionID":"00000000000000000000000000000000","ModTime":"2022-05-26T02:48:44.625903892-04:00","Signature":"b63cd20e","Type":1,"Flags":6},"Metadata":{"Type":1,"V2Obj":{"ID":"AAAAAAAAAAAAAAAAAAAAAA==","DDir":"D9JTg2uzSz+sKxN16Z7e1Q==","EcAlgo":1,"EcM":12,"EcN":4,"EcBSize":1048576,"EcIndex":11,"EcDist":[6,7,8,9,10,11,12,13,14,15,16,1,2,3,4,5],"CSumAlgo":1,"PartNums":[1],"PartETags":null,"PartSizes":[103794],"PartASizes":[103794],"Size":103794,"MTime":1653547724625903892,"MetaSys":{"x-minio-internal-inline-data":"dHJ1ZQ=="},"MetaUsr":{"etag":"71f3fcf7d01af8a9f1483ad512069e03","content-type":"image/jpg"}}}}]}
,
	"minio3:9000/data/disk2/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/xl.meta": {"Versions":[{"Idx":0,"Header":{"VersionID":"00000000000000000000000000000000","ModTime":"2022-05-26T02:48:44.625903892-04:00","Signature":"b63cd20e","Type":1,"Flags":6},"Metadata":{"Type":1,"V2Obj":{"ID":"AAAAAAAAAAAAAAAAAAAAAA==","DDir":"D9JTg2uzSz+sKxN16Z7e1Q==","EcAlgo":1,"EcM":12,"EcN":4,"EcBSize":1048576,"EcIndex":12,"EcDist":[6,7,8,9,10,11,12,13,14,15,16,1,2,3,4,5],"CSumAlgo":1,"PartNums":[1],"PartETags":null,"PartSizes":[103794],"PartASizes":[103794],"Size":103794,"MTime":1653547724625903892,"MetaSys":{"x-minio-internal-inline-data":"dHJ1ZQ=="},"MetaUsr":{"content-type":"image/jpg","etag":"71f3fcf7d01af8a9f1483ad512069e03"}}}}]}
,
	"minio2:9000/data/disk3/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/xl.meta": {"Versions":[{"Idx":0,"Header":{"VersionID":"00000000000000000000000000000000","ModTime":"2022-05-26T02:48:44.625903892-04:00","Signature":"b63cd20e","Type":1,"Flags":6},"Metadata":{"Type":1,"V2Obj":{"ID":"AAAAAAAAAAAAAAAAAAAAAA==","DDir":"D9JTg2uzSz+sKxN16Z7e1Q==","EcAlgo":1,"EcM":12,"EcN":4,"EcBSize":1048576,"EcIndex":15,"EcDist":[6,7,8,9,10,11,12,13,14,15,16,1,2,3,4,5],"CSumAlgo":1,"PartNums":[1],"PartETags":null,"PartSizes":[103794],"PartASizes":[103794],"Size":103794,"MTime":1653547724625903892,"MetaSys":{"x-minio-internal-inline-data":"dHJ1ZQ=="},"MetaUsr":{"content-type":"image/jpg","etag":"71f3fcf7d01af8a9f1483ad512069e03"}}}}]}
,
	"minio3:9000/data/disk3/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/xl.meta": {"Versions":[{"Idx":0,"Header":{"VersionID":"00000000000000000000000000000000","ModTime":"2022-05-26T02:48:44.625903892-04:00","Signature":"b63cd20e","Type":1,"Flags":6},"Metadata":{"Type":1,"V2Obj":{"ID":"AAAAAAAAAAAAAAAAAAAAAA==","DDir":"D9JTg2uzSz+sKxN16Z7e1Q==","EcAlgo":1,"EcM":12,"EcN":4,"EcBSize":1048576,"EcIndex":16,"EcDist":[6,7,8,9,10,11,12,13,14,15,16,1,2,3,4,5],"CSumAlgo":1,"PartNums":[1],"PartETags":null,"PartSizes":[103794],"PartASizes":[103794],"Size":103794,"MTime":1653547724625903892,"MetaSys":{"x-minio-internal-inline-data":"dHJ1ZQ=="},"MetaUsr":{"content-type":"image/jpg","etag":"71f3fcf7d01af8a9f1483ad512069e03"}}}}]}
,
	"minio2:9000/data/disk4/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/xl.meta": {"Versions":[{"Idx":0,"Header":{"VersionID":"00000000000000000000000000000000","ModTime":"2022-05-26T02:48:44.625903892-04:00","Signature":"b63cd20e","Type":1,"Flags":6},"Metadata":{"Type":1,"V2Obj":{"ID":"AAAAAAAAAAAAAAAAAAAAAA==","DDir":"D9JTg2uzSz+sKxN16Z7e1Q==","EcAlgo":1,"EcM":12,"EcN":4,"EcBSize":1048576,"EcIndex":3,"EcDist":[6,7,8,9,10,11,12,13,14,15,16,1,2,3,4,5],"CSumAlgo":1,"PartNums":[1],"PartETags":null,"PartSizes":[103794],"PartASizes":[103794],"Size":103794,"MTime":1653547724625903892,"MetaSys":{"x-minio-internal-inline-data":"dHJ1ZQ=="},"MetaUsr":{"etag":"71f3fcf7d01af8a9f1483ad512069e03","content-type":"image/jpg"}}}}]}
,
	"minio3:9000/data/disk4/ba-cam/2022/05/26/71288/18B02287_1_26052022_134811_tfwsC1.jpg/xl.meta": {"Versions":[{"Idx":0,"Header":{"VersionID":"00000000000000000000000000000000","ModTime":"2022-05-26T02:48:44.625903892-04:00","Signature":"b63cd20e","Type":1,"Flags":6},"Metadata":{"Type":1,"V2Obj":{"ID":"AAAAAAAAAAAAAAAAAAAAAA==","DDir":"D9JTg2uzSz+sKxN16Z7e1Q==","EcAlgo":1,"EcM":12,"EcN":4,"EcBSize":1048576,"EcIndex":4,"EcDist":[6,7,8,9,10,11,12,13,14,15,16,1,2,3,4,5],"CSumAlgo":1,"PartNums":[1],"PartETags":null,"PartSizes":[103794],"PartASizes":[103794],"Size":103794,"MTime":1653547724625903892,"MetaSys":{"x-minio-internal-inline-data":"dHJ1ZQ=="},"MetaUsr":{"content-type":"image/jpg","etag":"71f3fcf7d01af8a9f1483ad512069e03"}}}}]}

}`

@klauspost , I only see 8 shards. How are you able to say there are 9 shards ?

**Reply by klauspost** [reply]
Sure, 8. Still less than 12.

**Reply by mshanmu** [reply]
Makes sense, Thanks @klauspost
